File: CodeSolutions/169. Majority Element/169.java
// Language: Java
// Commit Timestamp: 2018-02-19T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

// Language: Java
// Commit Timestamp: 2018-02-19T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

// Language: Java
// Commit Timestamp: 2018-02-19T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

class Solution {
  public int majorityElement(int[] nums) {
    Integer ans = null;
    int count = 0;

    for (final int num : nums) {
      if (count == 0)
        ans = num;
      count += num == ans ? 1 : -1;
    }

    return ans;
  }
}

File: CodeSolutions/169. Majority Element/169.py
// Language: Java
// Commit Timestamp: 2018-01-20T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

// Language: Java
// Commit Timestamp: 2018-01-20T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

// Language: Java
// Commit Timestamp: 2018-01-20T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

class Solution:
  def majorityElement(self, nums: list[int]) -> int:
    ans = None
    count = 0

    for num in nums:
      if count == 0:
        ans = num
      count += (1 if num == ans else -1)

    return ans
